Ical.Net.Tests/CalendarEventTest.cs

```diff
@@ -472,78 +475,99 @@ public void HourMinuteSecondOffsetParsingTest()
 
 
     [Test, Category("CalendarEvent")]
-    public void TestGetEffectiveDuration()
+    public void GetNominalDurationTests()
     {
-        var now = _now.Subtract(TimeSpan.FromTicks(_now.Ticks % TimeSpan.TicksPerSecond));
+        var dt = new DateTime(2025, 3, 1, 14, 30, 0);
+        const string tzIdStart = "America/New_York";
+        const string tzIdEnd = "Europe/London";
 
-        var evt = new CalendarEvent()
+        var evt = new CalendarEvent
         {
-            DtStart = new CalDateTime(DateOnly.FromDateTime(now), TimeOnly.FromDateTime(now)),
-            DtEnd = new CalDateTime(DateOnly.FromDateTime(now.AddHours(1)), TimeOnly.FromDateTime(now.AddHours(1)))
+            DtStart = new CalDateTime(DateOnly.FromDateTime(dt), TimeOnly.FromDateTime(dt), tzIdStart),
+            DtEnd = new CalDateTime(DateOnly.FromDateTime(dt.AddHours(1)), TimeOnly.FromDateTime(dt.AddHours(1)), tzIdEnd)
         };
 
         Assert.Multiple(() =>
         {
-            Assert.That(evt.DtStart.Value, Is.EqualTo(now));
-            Assert.That(evt.DtEnd.Value, Is.EqualTo(now.AddHours(1)));
-            Assert.That(evt.GetFirstDuration(), Is.EqualTo(TimeSpan.FromHours(1)));
+            Assert.That(evt.DtStart.Value, Is.EqualTo(dt));
+            Assert.That(evt.DtEnd.Value, Is.EqualTo(dt.AddHours(1)));
+            Assert.That(evt.CalcFirstNominalDuration(), Is.EqualTo(TimeSpan.FromHours(1)));
```

Comment:
It should not be possible to calculate a nominal duration if the tzids don't match as the value usually doesn't make much sense.